### PR TITLE
[JIT] Optimize before inlining

### PR DESCRIPTION
**Pull request by eellison**
This speeds up the inlining pass of FairSeq model from 180s -> 13s.

### CODE REVIEW

File: torch/csrc/jit/api/function_impl.cpp [hunk 1]
```diff
@@ -1,7 +1,8 @@
 #include <torch/csrc/jit/api/function_impl.h>
-#include <torch/csrc/jit/passes/inliner.h>
-
 #include <torch/csrc/jit/frontend/error_report.h>
+#include <torch/csrc/jit/passes/inliner.h>
+#include <torch/csrc/jit/passes/peephole.h>
+#include "torch/csrc/jit/passes/constant_propagation.h"
```
Comment (ZolotukhinM) [inline torch/csrc/jit/api/function_impl.cpp]: Nit: use <> for includes.

**Review by ZolotukhinM — APPROVED** [review]
Awesome! I planned to do it a long time ago and am so happy to see it finally happening :)

File: torch/csrc/jit/api/function_impl.cpp (hunk 1 above)
Comment (ZolotukhinM) [reply torch/csrc/jit/api/function_impl.cpp]: Also, I think clang-format sorts them and currently they are not sorted.

**Comment by eellison** [comment]
@ZolotukhinM thanks for doing all the work!

File: torch/csrc/jit/api/function_impl.cpp (hunk 1 above)
Comment (eellison) [reply torch/csrc/jit/api/function_impl.cpp]: this is with clang-format...

**Pull request closed without merging**